java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/hash/SimpleHasher.java

```diff
@@ -110,7 +111,7 @@ public int hashCode() {
   }
 
   @Override
-  public boolean equals(Object obj) {
+  public boolean equals(@Nullable Object obj) {
```

Comment:
Just curious: should we annotate equals method with Nullable?

Reply:
If it is certain that it will never be null, then any annotations need to be added. When there are logic/circuit/steps that can be null or not, then Nullable is needed (that does not mean that it is always null).

Reply:
That's an interesting thread 👍